lib/components/steps/mapping/MappingTab.tsx

```diff
@@ -248,6 +261,58 @@ const MappingTab = ({defaultCollection}: MappingProps) => {
         return false
     };
 
+    const isSameStrategyType = (filter: SortingStrategy, newFilter: SortingStrategy) => {
+        if (!filter || !newFilter) {
+            return false;
+        }
+        return filter.constructor === newFilter.constructor;
+    }
+
+    const sortRows = useCallback((fitlerStrategy: SortingStrategy) => {
+        const result = fitlerStrategy.doSort(visibleRows, datasetMapping, headerIndexes);
+        setVisibleRows([...result]);
+    }, [datasetMapping, headerIndexes, visibleRows])
+
+    const handleSortOrderChange = useCallback((newSortOrder: number) => {
+        if (currentFilterStrategy instanceof StatusFilter) {
+            setSortOrder({
+                ...sortOrder,
+                statusSortOrder: newSortOrder
+            })
+        } else if (currentFilterStrategy instanceof VariableNameFilter) {
+            setSortOrder({
+                ...sortOrder,
+                variableNameSortOrder: newSortOrder
+            });
+        } else if (currentFilterStrategy instanceof CdeSortingFilter) {
+            setSortOrder({
+                ...sortOrder,
+                cdeSortOrder: newSortOrder
+            });
+        }
```

Comment:
Issue (non-blocking): I don't like this if and else logic, it sort of goes against the idea of using the strategy pattern here. I think my suggestion for having this complex state variable might have not been the best. What if we try to have just 1 simple sortOrder state variable like I initially proposed and in the sortIcon we do something like:
```
<SortIcon direction={currentFilterStrategy instanceof XSortingFilter ? sortOrder state variable : 0}/>
```